test/unit/user_test.rb

```diff
@@ -119,4 +119,27 @@ def test_create_user_successful_service_not_require_registered_users
 
     assert_equal metrics, user.load_metric_names
   end
+
+  test '.delete_all removes all the Users of a Service' do
+    Service.save!(id: 7003, provider_key: 'test_provkey', default_service: false)
+    Service.save!(id: 7004, provider_key: 'test_provkey', default_service: false)
+    num_users = 600
+    for i in 1..num_users
+      User.save! username: "username#{i}", service_id: 7003, plan_id: '1001',
+                 plan_name: 'planname'
+    end
+    User.save! username: "username_differentservice", service_id: 7004,
+               plan_id: '1001',plan_name: 'planname'
+
+    User.delete_all(7003)
+
+    for i in 1..num_users
```

Comment:
Nitpick: In Ruby `(1..num_users).each { |n| ... }` is more idiomatic.

Reply:
maybe `1.upto(num_users).each { |n| ... }` ?? just another option, valid as well.

Reply:
If say that the only allowed option here is #times 😆

Reply:
but `#times` is 0-based. If index is needed and 1-based as its first implementation, I'd say it is better

`1.upto(num_users).each { |n| ... }`